brainscore_vision/data_helpers/s3.py

```diff
@@ -0,0 +1,72 @@
+"""
+Will contain functions for loading data assemblies and stimuli sets from s3,
+and whatever helpers are needed.
+
+Should provide similar functionality to brainio/lookup.py and brainio/fetch.py,
+but without use of lookup.csv
+"""
+import functools
+import logging
+import os
+from pathlib import Path
+from typing import Callable
+
+from brainio.assemblies import DataAssembly, AssemblyLoader, StimulusMergeAssemblyLoader
+from brainio.fetch import fetch_file, unzip, resolve_stimulus_set_class
+from brainio.stimuli import StimulusSetLoader, StimulusSet
+
+_logger = logging.getLogger(__name__)
+
+
+def get_path(identifier: str, file_type: str, bucket: str, version_id: str, sha1: str):
+    """
+    Finds path of desired file (for .csvs, .zips, and .ncs).
+    """
+    # for stimuli sets
+    if file_type == 'csv' or file_type == 'zip':
+        assembly_prefix = 'image_'
+    # for data assemblies
+    else:
+        assembly_prefix = 'assy_'
+
+    filename = f"{assembly_prefix}{identifier.replace('.', '_')}.{file_type}"
+    file_path = fetch_file(location_type="S3",
+                           location=f"https://{bucket}.s3.amazonaws.com/{filename}",
+                           version_id=version_id,
+                           sha1=sha1)
+    return file_path
+
+
+def load_assembly_from_s3(identifier: str, version_id: str, sha1: str, bucket: str, cls: type,
+                          stimulus_set_loader: Callable[[], StimulusSet] = None) -> DataAssembly:
+    file_path = get_path(identifier, 'nc', bucket, version_id, sha1)
+    if stimulus_set_loader:  # merge stimulus set meta into assembly if `stimulus_set_loader` is passed
+        stimulus_set = stimulus_set_loader()
+        loader_class = functools.partial(StimulusMergeAssemblyLoader,
```

Comment:
should make this optional -- pass in flag to merge stimulus set meta into assembly or not